patrickbrett/mlvet
# Provide demo usage of undo stack

~~Depends on: https://github.com/patrickbrett/mlvet/pull/23 (line count will drop to like +250 when that one gets merged)~~ 

Provides demo usage of undo stack, with two dummy ops:
- delete half the words in the transcript
- replace a random word with 'swamp'

To test:
- open a project
- press the op buttons (delete half the words, replace a word with swamp) in various combinations
- check that the undo and redo react buttons work
- check that using ctrl+z / ctrl+shift+z also works
- check that the edit menu buttons work
- check that they're greyed out at the same time as the react buttons
- check that undoing a few times, then hitting an op button again (e.g. delete half the words), wipes out the option to redo the things that you undid
- check that no matter what you do, if you hit undo enough you'll end back with the initial state of the transcription

While you're at it feel free to check out the redux state from the debugger if you wanna play with it live, I find it kind of satisfying lol
